src/server/ua_server.c

```diff
@@ -167,6 +167,14 @@ void UA_Server_delete(UA_Server *server) {
     }
     UA_Array_delete(server->namespaces, server->namespacesSize, &UA_TYPES[UA_TYPES_STRING]);
 
+#ifdef UA_ENABLE_FILETYPE_OBJECT_SUPPORT
+    UA_FileType *fileObject, *fileObject_tmp;
+    LIST_FOREACH_SAFE(fileObject, &server->fileObjects, listEntry, fileObject_tmp)
+    {
+        UA_Server_deleteNode(server, fileObject->fileNodeId, true);
```

Comment:
This will likely cause the   UA_assert(++(lock->mutexCounter) == 1); in UA_LOCK which is called in UA_Server_deleteNode to fail.